lib/receptor_controller/client/response_worker.rb

```diff
@@ -43,13 +60,19 @@ def stop
     # @param receiver [Object] any object implementing callbacks
     # @param response_callback [Symbol] name of receiver's method processing responses
     # @param timeout_callback [Symbol] name of receiver's method processing timeout [optional]
-    def register_message(msg_id, receiver, response_callback: :response_received, timeout_callback: :response_timeout)
-      registered_messages[msg_id] = {:receiver => receiver, :response_callback => response_callback, :timeout_callback => timeout_callback, :registered_at => Time.now.utc}
+    # @param error_callback [Symbol] name of receiver's method processing errors [optional]
+    def register_message(msg_id, receiver, response_callback: :response_success, timeout_callback: :response_timeout, error_callback: :response_error)
```

Comment:
(more thoughts on next steps with redis)

this method will probably get moved to taking in an array of hashes in the form of `{:name => "availability_check", :action => &block}` which gets passed in from the ansible tower client (or whatever is implementing, that way we can map whatever action we're requesting (ie availability_check) to a function that does what we want (ie update source after return). 

That way it will all be statically defined - the worker just pulls messages off the queue and does the work defined by the return-message's action.